server/plugin/webhook_test.go

```diff
@@ -0,0 +1,869 @@
+package plugin
+
+import (
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/google/go-github/v54/github"
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+func TestPostPushEvent(t *testing.T) {
+	mockKvStore, mockAPI, _, _, _ := GetTestSetup(t)
+	p := getPluginTest(mockAPI, mockKvStore)
+
+	tests := []struct {
+		name      string
+		pushEvent *github.PushEvent
+		setup     func()
+	}{
+		{
+			name:      "no subscription found",
+			pushEvent: GetMockPushEvent(),
+			setup: func() {
+				mockKvStore.EXPECT().Get(SubscriptionsKey, gomock.Any()).Return(nil).Times(1)
+			},
+		},
+		{
+			name:      "no commits found in event",
+			pushEvent: GetMockPushEventWithoutCommit(),
+			setup: func() {
+				mockKvStore.EXPECT().Get(SubscriptionsKey, gomock.Any()).DoAndReturn(func(key string, value interface{}) error {
+					if v, ok := value.(**Subscriptions); ok {
+						*v = GetMockSubscriptions()
+					}
+					return nil
+				}).Times(1)
+			},
+		},
+		{
+			name:      "Error creating post",
+			pushEvent: GetMockPushEvent(),
+			setup: func() {
+				mockKvStore.EXPECT().Get(SubscriptionsKey, gomock.Any()).DoAndReturn(func(key string, value interface{}) error {
+					if v, ok := value.(**Subscriptions); ok {
+						*v = GetMockSubscriptions()
+					}
+					return nil
+				}).Times(1)
+				mockAPI.On("CreatePost", mock.Anything).Return(nil, &model.AppError{Message: "error creating post"}).Times(1)
+				mockAPI.On("LogWarn", "Error webhook post", "post", mock.Anything, "error", "error creating post")
+			},
+		},
+		{
+			name:      "Successful handle post push event",
+			pushEvent: GetMockPushEvent(),
+			setup: func() {
+				mockKvStore.EXPECT().Get(SubscriptionsKey, gomock.Any()).DoAndReturn(func(key string, value interface{}) error {
+					if v, ok := value.(**Subscriptions); ok {
+						*v = GetMockSubscriptions()
+					}
+					return nil
+				}).Times(1)
+				mockAPI.On("CreatePost", mock.Anything).Return(&model.Post{}, nil).Times(1)
+			},
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.setup()
+
+			p.postPushEvent(tc.pushEvent)
+
+			mockAPI.AssertExpectations(t)
+		})
+	}
+}
+
+func TestPostCreateEvent(t *testing.T) {
+	mockKvStore, mockAPI, _, _, _ := GetTestSetup(t)
+	p := getPluginTest(mockAPI, mockKvStore)
+
+	tests := []struct {
+		name        string
+		createEvent *github.CreateEvent
+		setup       func()
+	}{
+		{
+			name:        "no subscription found",
```

Comment:
```suggestion
			name:        "No subscription found",
```
For consistency, check everywhere